Assets/Scripts/Utils/NetworkSimulatorUIMediator.cs

```diff
@@ -91,11 +95,41 @@ void OnPresetChanged(int optionIndex)
         {
             m_NetworkSimulator.ChangeConnectionPreset(m_SimulatorPresets[m_PresetsDropdown.options[optionIndex].text]);
         }
+
+        void OnScenarioChanged(int optionIndex)
+        {
+            var scenarioName = m_ScenariosDropdown.options[optionIndex].text;
+            NetworkScenario scenario = null;
+            switch (scenarioName)
+            {
+                case k_None:
+                    break;
+                case k_ConnectionCyclesScenarioName:
+                    scenario = new ConnectionsCycle();
+                    ((ConnectionsCycle) scenario).Configurations.Add(new ConnectionsCycle.Configuration() {ChangeIntervalMilliseconds = 5000, ConnectionPreset = NetworkSimulatorPresets.HomeBroadband});
+                    ((ConnectionsCycle) scenario).Configurations.Add(new ConnectionsCycle.Configuration() {ChangeIntervalMilliseconds = 5000, ConnectionPreset = NetworkSimulatorPresets.Mobile5G});
+                    break;
+                case k_RandomConnectionSwapScenarioName:
+                    scenario = new RandomConnectionsSwap();
+                    ((RandomConnectionsSwap) scenario).ChangeIntervalMilliseconds = 5000;
+                    ((RandomConnectionsSwap) scenario).Configurations.Add(new RandomConnectionsSwap.Configuration() {ConnectionPreset = NetworkSimulatorPresets.HomeBroadband});
+                    ((RandomConnectionsSwap) scenario).Configurations.Add(new RandomConnectionsSwap.Configuration() {ConnectionPreset = NetworkSimulatorPresets.Mobile5G});
```

Comment:
**Suggestion:** Ideally you should avoid casting scenario to `RandomConnectionSwap` or `ConnectionsCycle` so many times, introducing a variable would be preferrable, perhaps something like this:

```
case k_ConnectionCyclesScenarioName:
    var connectionsCycle = new ConnectionsCycle();
    connectionsCycle.Configurations.Clear();
    foreach (var configuration in m_ConnectionsCycleConfigurations)
    {
        connectionsCycle.Configurations.Add(configuration);
    }
    scenario = connectionsCycle;
    m_PresetsDropdown.captionText.color = m_PresetsDropdown.colors.disabledColor;
    m_PresetsDropdown.interactable = false;
    break;
case k_RandomConnectionSwapScenarioName:
    var randomConnectionsSwap = new RandomConnectionsSwap();
    randomConnectionsSwap.Configurations.Clear();
    foreach (var configuration in m_RandomConnectionsSwapConfigurations)
    {
        randomConnectionsSwap.Configurations.Add(configuration);
    }
    scenario = randomConnectionsSwap;
    m_PresetsDropdown.captionText.color = m_PresetsDropdown.colors.disabledColor;
    m_PresetsDropdown.interactable = false;
    break;
```

Reply:
Good point, I'll fix this